pkg/agent/client.go

```diff
@@ -351,62 +354,73 @@ func (a *Client) Serve() {
 
 		switch pkt.Type {
 		case client.PacketType_DIAL_REQ:
-			klog.V(4).Infoln("received DIAL_REQ")
-			resp := &client.Packet{
-				Type:    client.PacketType_DIAL_RSP,
-				Payload: &client.Packet_DialResponse{DialResponse: &client.DialResponse{}},
-			}
+			go func() {
```

Comment:
If we move the entire handling of DIAL_REQ into a goroutine like this, we can still enter a race with the handling of CLOSE_REQ that I mentioned in https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/179#issuecomment-839283626.

How about we only move the net.Dial into a goroutine? That way `a.connManager.Add` is called sequentially and that would prevent the race. This is also what I suggested in https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/179#issuecomment-841347676.

Reply:
@caesarxuchao So what you are suggesting is that the go routine should start after the `a.connManager.Add` call. Right?
In the beginning the go routine simply started shortly before the `net.Dial` call, but after your suggested refactoring we can indeed move the start of the go routine to a later point in time, i.e. directly before the `net.Dial` call.

Reply:
Yeah, the goroutine should start after the `a.connManager.Add` call. Or in other words, "move the start of the go routine to a later point in time, i.e. directly before the net.Dial call".